### PR TITLE
Added the resize quick button

### LINKED ISSUE #499: Add resize module to list of quick buttons
This issue is a sub-issue of #487 , which is to be broken down into smaller issues.

In the image-sequencer, there are a list of quick-buttons so the user can add commonly-used layers quickly:
image

What we want to do is to add another quick-add button for the resize functionality. Here's the todo for this task:

- [ ] Find a suitable icon on font-awesome and link it here **make sure it's from the v4.5 icon set!**
- [ ] Add the icon in the appropriate button style in examples/index.html.
image
- [ ] Keep in mind the proper syntax for data-value, which can be checked by navigating to the proper component in src/modules. It is lowercase

### Please show us where to look

http://sequencer.publiclab.org/examples/#steps=

### What's your PublicLab.org username?

JonathanXu1

### Browser, version, and operating system

Firefox63, Windows 10

****

## Thank you!

Your help makes Public Lab better! We *deeply* appreciate your helping refine and improve this site.

To learn how to write really great issues, which increases the chances they'll be resolved, see:

https://publiclab.org/wiki/developers#Contributing+for+non-coders

### PR DESCRIPTION
**Pull request by Harshithpabbati**
Fixes #499 

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `npm test`
* [x] code is in uniquely-named feature branch and has no merge conflicts
* [x] PR is descriptively titled
* [x] ask `@publiclab/reviewers` and `@publiclab/is-reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software

Thanks!

### CODE REVIEW

**Comment by Harshithpabbati** [comment]
@publiclab/is-reviewers

**Comment by Harshithpabbati** [comment]
@HarshKhandeparkar checkout this..

File: examples/index.html [hunk 1]
```diff
@@ -186,19 +192,36 @@ <h2 style="margin-top:20px">Save</h2>
     </section>
     </div>
   </div>
-
+  <button id="move-up"><i class="fa fa-arrow-circle-o-up"></i></button>
+</body>
   <footer>
     <hr style="margin:20px;"><center><a class="color:grey;" id="clear-cache">Clear offline cache</a></center>
+    <div class="row">
```
Comment (HarshKhandeparkar) [inline examples/index.html]: I think you have made this branch from your footer branch and not from main. Please remove these changes(footer)

File: examples/index.html [hunk 2]
```diff
@@ -86,6 +86,12 @@ <h1><a href="/" target='_blank' class="name-header">Image Sequencer</a></h1>
           </div>
           <p class="info">Select a new module to add to your sequence.</p>
           <div class="row center-align radio-group">
+            <div>
+              <div class="radio" data-value="resize">
+                <i class="icon-resize-full fa-4x i-over"></i>
```
Comment (HarshKhandeparkar) [inline examples/index.html]: You have to add the classes `fa fa-4x i-over` and `fa-name-of-icon`. Do you have any other difficulty?

File: examples/lib/intermediateHtmlStepUi.js [hunk 3]
```diff
@@ -10,6 +10,12 @@ function IntermediateHtmlStepUi(_sequencer, step, options) {
             <div class="panel-body">\
               <p class="info">Select a new module to add to your sequence.</p>\
               <div class="row center-align radio-group">\
+              <div>\
+              <div class="radio" data-value="resize">\
+              <i class="fa icon-resize-full fa-4x"></i>\
```
Comment (HarshKhandeparkar) [inline examples/lib/intermediateHtmlStepUi.js]: Also here.

**Comment by HarshKhandeparkar** [comment]
Also you have to build the dist riles by first running `grunt browserify` and then `grunt uglify` and please commit those files.

**Comment by Divy123** [comment]
@Harshithpabbati please provide a screenshot or gif .
Also as said by HarshKhandeparkar please checkout main branch then make a new branch suppose add-resize and make the said changes there.
Please keep this rule with you new PR by new branch always.
Thanks!

**Comment by HarshKhandeparkar** [comment]
@Divy123 @Harshithpabbati I don't think there is a need to checkout main, just remove the footer.

**Comment by Divy123** [comment]
@HarshKhandeparkar this a rule which I said that new PR by new branch.

**Comment by HarshKhandeparkar** [comment]
Divy123 this is a new branch, it is just branched from the footer branch so it has those changes.

**Comment by Harshithpabbati** [comment]
I will remove the footer and send it to u

**Comment by Divy123** [comment]
Also @Harshithpabbati something I want to know from you that if you have created a new branch resize by remaining on footer branch.
I think that may be the case because of that only your footer changes are visible.
So the order is always checkout main from your current branch before making changes that will result into a new PR

**Comment by Harshithpabbati** [comment]
Yeah i did it by mistake i didn't see it

**Comment by Divy123** [comment]
@Harshithpabbati donl't do this as this is not a good practice.
Please checkout to the main and then do checkout a new branch so that no any unexpected changes may happen.

**Comment by HarshKhandeparkar** [comment]
@Harshithpabbati when you create a new branch, it is branched from the branch that is currently checked out. Always checkout main before branching. Also keep main up to date. If you are using the git tab of a code editor like atom or vscode then it normally asks which branch to branch from. Always choose HEAD branch. Thanks.

**Comment by jywarren** [comment]
This is looking great! I just needed resize recently while working with an
image and thought how useful a quick button would be.

Don't worry you'll get the hang of branch management sooner enough, it's a
pain at first but you get used to it. Thanks to everyone for helping out,
and thanks for contributing this code! 👍🏽👍🏽🙌🏽

**Comment by Harshithpabbati** [comment]
How do I edit it??Do I close the request and send a new pr??

**Comment by HarshKhandeparkar** [comment]
You can commit to the branch and the changes will be visible here.

**Comment by Harshithpabbati** [comment]
even though i added the font awesome icons i can't get the icon

**Comment by HarshKhandeparkar** [comment]
Can you please commit the changes?

**Comment by Harshithpabbati** [comment]
@HarshKhandeparkar made the commit

**Comment by Harshithpabbati** [comment]
screenshot from 2019-02-16 20-47-20
This was the changes which I made.But can't get the image in it's place

**Comment by HarshKhandeparkar** [comment]
I don't think this is your problem. @Divy123 the preview is not loading. Can you please look into it?

**Comment by Harshithpabbati** [comment]
> I don't think this is your problem. @Divy123 the preview is not loading. Can you please look into it?

screenshot from 2019-02-16 20-59-45
Checkout this @HarshKhandeparkar

**Comment by HarshKhandeparkar** [comment]
Yes, exactly the preview image behind the icon is not loading. @Divy123 please look into this.

**Comment by Divy123** [comment]
@Harshithpabbati you need to do just one thing.
Navigate to the `image-sequencer/examples/lib/insertPreview` file and at the line number 33 inside ` previewSequencerSteps' you need to add resize:"125% .
I think that should do.

**Comment by Harshithpabbati** [comment]
> @Harshithpabbati you need to do just one thing.
> Navigate to the `image-sequencer/examples/lib/insertPreview` file and at the line number 33 inside `previewSequencerSteps`
> you need to add resize:"125%" .
> I think that should do.
> Thanks !
> Please feel free to ask in case you need any help.

thanks for the help @Divy123

**Comment by Harshithpabbati** [comment]
screenshot from 2019-02-16 21-54-06
Done with it
@jywarren please review it :)

**Comment by Divy123** [comment]
> Yes, exactly the preview image behind the icon is not loading. @Divy123 please look into this.

@HarshKhandeparkar the preview has to be added specifically to the above mentioned object as it loads the previews.
I hope that's clear :smile:

File: examples/lib/intermediateHtmlStepUi.js [hunk 4]
```diff
@@ -10,6 +10,12 @@ function IntermediateHtmlStepUi(_sequencer, step, options) {
             <div class="panel-body">\
               <p class="info">Select a new module to add to your sequence.</p>\
               <div class="row center-align radio-group">\
+              <div>\
+              <div class="radio" data-value="resize">\
+              <i class="fa fa-icon-resize-full fa-4x"></i>\
```
Comment (Divy123) [inline examples/lib/intermediateHtmlStepUi.js]: use of i-over class here.

File: examples/index.html [hunk 5]
```diff
@@ -86,6 +86,12 @@ <h1><a href="/" target='_blank' class="name-header">Image Sequencer</a></h1>
           </div>
           <p class="info">Select a new module to add to your sequence.</p>
           <div class="row center-align radio-group">
+            <div>
+              <div class="radio" data-value="resize">
+                <i class="fa fa-arrows-alt fa-4x"></i>
```
Comment (Divy123) [inline examples/index.html]: Use of i-over class should happen here!

File: examples/demo.css [hunk 6]
```diff
@@ -106,7 +106,7 @@ h1 {
 }
 
 #addStep {
-  max-width: 500px;
+  max-width: 5000px;
```
[the remark this answers is not in the document]
Comment (HarshKhandeparkar) [reply examples/demo.css]: There is no need to increase max-width. The overflow is handled by adding a scrollbar(x-axis)

[the remark this answers is not in the document]
Comment (Harshithpabbati) [reply examples/demo.css]: it looks good with out the scrollbar.

[the remark this answers is not in the document]
Comment (HarshKhandeparkar) [reply examples/demo.css]: It goes out of the page on some screen sizes. That's why the scrollbar was added in the first place

[the remark this answers is not in the document]
Comment (HarshKhandeparkar) [reply examples/demo.css]: It goes out of the page on some screen sizes. That's why the scrollbar was added in the first place

**Comment by Harshithpabbati** [comment]
@publiclab/is-reviewers

**Comment by Harshithpabbati** [comment]
Done with all the changes.

**Comment by Divy123** [comment]
@Harshithpabbati screenshot please.

**Comment by Harshithpabbati** [comment]
screenshot from 2019-02-16 23-00-46

**Comment by Divy123** [comment]
Please do grunt build

**Comment by HarshKhandeparkar** [comment]
Please run `grunt browserify` and `grunt uglify` and commit the changes.

**Comment by HarshKhandeparkar** [comment]
@Divy123 currently grunt build does not build the ui files.

**Comment by Divy123** [comment]
Oh my bad!
Thanks @HarshKhandeparkar .

**Comment by Harshithpabbati** [comment]
> Please run `grunt browserify` and `grunt uglify` and commit the changes.

Yeah i did it before using npm start

**Comment by HarshKhandeparkar** [comment]
You have yo commit the files inside dist/ folder after running the grunt commands. You have to run the grunt commands each time you touch a js file. Thanks.

**Comment by Harshithpabbati** [comment]
@publiclab/is-reviewers done with the changes.

**Comment by Harshithpabbati** [comment]
@publiclab/is-reviewers

**Comment by HarshKhandeparkar** [comment]
Your PR has been reviewed already and is ready to merge. When jywarren sees your PR next time , he will merge it(after a final review). According to his time, now will be the beginning of a sunday so you will have to wait for another 24hrs before it gets merged. Thanks!

**Comment by Harshithpabbati** [comment]
Oh ok fine

**Comment by Harshithpabbati** [comment]
I have a doubt how can i join into the reviewers team??

**Comment by HarshKhandeparkar** [comment]
Jywarren is really busy since he has to look after all the projects of publiclab. The biggest project i.e. plots2 has more than 100 PRs open so its really difficult for him to reply to all the questions. I hope you understand.

**Comment by HarshKhandeparkar** [comment]
> I have a doubt how can i join into the reviewers team??

There are some prerequisites mentioned in #656 and you have to be well acquainted with the codebase and have to have some experience working with the publiclab community.

**Comment by jywarren** [comment]
Great work here!!! Thanks so much!

**Comment by jywarren** [comment]
And we'd love to have more contributions from you! 🎉🙌🏽

**Comment by Harshithpabbati** [comment]
> And we'd love to have more contributions from you! 🙌🏽

Sure @jywarren 
Can i be a part of reviewers team :)

**Pull request merged**